fitbenchmarking/fitbenchmarking
# Merging release branch into master

#### Description of Work

Fixes #536, Fixes #532, Fixes #196, Fixes #511, Fixes #523, Fixes #543 

Updates master branch with all the PR which were merged into the release branch

#### Testing Instructions

1. Check all tests run.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
